docs/authorization/globus_app/apps.rst

```diff
@@ -0,0 +1,89 @@
+.. _globus_apps:
+
+.. currentmodule:: globus_sdk
+
+GlobusApps
+==========
+
+
+.. warning::
+
+    Currently ``GlobusApp`` only supports scripting-based use cases.
+
+    Hosted applications (i.e. web servers) should see :ref:`globus_authorizers` instead.
+
+
+:class:`GlobusApp` is a high level construct designed to simplify the process of
+of authenticating for Globus service clients.
+
+A ``GlobusApp`` will always require a programmatic entity called a "client". Depending
+on the type of ``GlobusApp`` in use, the client will play different roles in the
+request flow.
```

Comment:
```suggestion
A ``GlobusApp`` requires a programmatic entity called a "client". Depending
on the type of ``GlobusApp`` in use, the client will play different roles in the
request flow.
```

Reply:
> Wondering if the first use of client here needs to link out to the relevant Auth docs: https://docs.globus.org/api/auth/developer-guide/#developing-apps

I vote (strongly) against this. We should have SDK-specific documentation on this subject, and having two "sources of truth" for this information will confuse users.

> ...also wondering if we should indicate which of the options from the provided list "Service Account, Portal / Science Gateway, Thick Client, API, or Advanced Application" are relevant?

:+1: for this.
I have some minor gripes about those categories being what they are[^1], but I think matching the web experience in our docs will help users.

[^1]: This part of the webapp is only for developers, but intentionally choosing use-case names rather than names meant to match the SDKs and Auth docs _assumes_ that the user in question is non-technical. It's dissonant and I don't see that it was necessary to introduce this discrepancy. I strongly believe that this is a poor UX decision, but I only minorly care about it.

Reply:
I'm totally okay with not doing the first suggestion if we do the second 😄 on reflection, I think that's actually the one I care more about.

Reply:
Refactored this section a good amount; the linkage to specific webapp based terminology (point two) is included now in the comparison table.